File: app/src/main/java/com/amazon/tv/leanbacklauncher/recommendations/GservicesRankerParameters.java
package com.amazon.tv.leanbacklauncher.recommendations;

import android.content.ContentResolver;
import android.content.Context;

import com.amazon.tv.tvrecommendations.service.RankerParameters;
import com.amazon.tv.tvrecommendations.service.RankerParametersFactory;

final class GservicesRankerParameters extends RankerParameters {
    private ContentResolver mContentResolver;

    public static class Factory implements RankerParametersFactory {
        public RankerParameters create(Context context) {
            return new GservicesRankerParameters(context);
        }
    }

    private GservicesRankerParameters(Context context) {
        this.mContentResolver = context.getApplicationContext().getContentResolver();
    }

    public Object getVersionToken() {
        return 0;
    }

    public float getFloat(String key, float defaultValue) {
        return defaultValue;
    }
}

File: app/src/main/java/com/amazon/tv/leanbacklauncher/recommendations/GservicesRankerParameters.kt
package com.amazon.tv.leanbacklauncher.recommendations

import android.content.Context
import com.amazon.tv.tvrecommendations.service.RankerParameters
import com.amazon.tv.tvrecommendations.service.RankerParametersFactory

internal class GservicesRankerParameters private constructor() :
    RankerParameters() {
    class Factory : RankerParametersFactory {
        override fun create(context: Context): RankerParameters {
            return GservicesRankerParameters()
        }
    }

    public override fun getVersionToken(): Any {
        return 0
    }

    public override fun getFloat(key: String, defaultValue: Float): Float {
        return defaultValue
    }

}